src/main/java/com/box/sdk/BoxFile.java

```diff
@@ -1913,7 +1913,10 @@ protected void parseJSONMember(JsonObject.Member member) {
         }
 
         private BoxFileVersion parseFileVersion(JsonObject jsonObject) {
-            return new BoxFileVersion(BoxFile.this.getAPI(), jsonObject, BoxFile.this.getID());
+            if (!jsonObject.isEmpty()) {
+                return new BoxFileVersion(BoxFile.this.getAPI(), jsonObject, BoxFile.this.getID());
+            }
+            return null;
```

Comment:
For the sake of clarity I suggest that we deal with fail cases first. More about this pattern here https://medium.com/swlh/return-early-pattern-3d18a41bba8

```suggestion
            if (jsonObject.isEmpty()) {
                return null;                
            }
            return new BoxFileVersion(BoxFile.this.getAPI(), jsonObject, BoxFile.this.getID());
```

Reply:
Some of my Sign Requests do not have `file_version` in `files` located in `source_files`. It is causing an error during deserialisation. Contacted SignAPI team to check if that's intended. I'm not sure how to reproduce it, though.

Reply:
Confirmed that it was a bug before, but it should be already fixed. It should be safe to remove this line now.